packages/seo-integration/src/analyze.js

```diff
@@ -0,0 +1,144 @@
+import { Paper } from "yoastseo";
+
+/**
+ * Creates a paper based on the given data, keyphrase and configuration.
+ *
+ * @param {Object} data The paper data and metadata, e.g. content, title, date, slug.
+ * @param {Object} keyphrase The keyphrase to use for the analysis.
+ * @param {Object} configuration Extra configuration.
+ *
+ * @returns {Paper} A paper that can be analyzed using the web worker.
+ */
+function createPaper( data, keyphrase, configuration ) {
+	return new Paper(
+		data.content,
+		{
+			// Keyphrase data.
+			keyword: keyphrase.keyphrase,
+			synonyms: keyphrase.synonyms,
+			// General data and metadata.
+			description: data.metaDescription,
+			title: data.seoTitle,
+			titleWidth: data.seoTitleWidth,
+			permalink: data.slug,
+			date: data.date,
+			// Configuration data.
+			locale: configuration.locale,
+		},
+	);
+}
+
+/**
+ * Adapts the object with related keyphrases to a a structure
+ * that the analysis web worker is able to consume.
+ *
+ * @param {Object} relatedKeyphrases The related keyphrases.
+ *
+ * @returns {Object} The transformed related keyphrases.
+ */
+function adaptRelatedKeyphrases( relatedKeyphrases ) {
+	const transformed = {};
+
+	Object.keys( relatedKeyphrases ).forEach(
+		id => {
+			transformed[ id ] = {
+				keyword: relatedKeyphrases[ id ].keyphrase,
+				synonyms: relatedKeyphrases[ id ].synonyms,
+			};
+		},
+	);
+
+	return  transformed;
```

Comment:
This could be a one-liner function with the use of the [mapValues](https://lodash.com/docs/4.17.15#mapValues) from Lodash:

mapValues( relatedKeyphrases, ( { keyphrase: keyword, synonyms } ) => ( { keyword, synonyms } ) )
```suggestion
	return mapValues( relatedKeyphrases, ( { keyphrase: keyword, synonyms } ) => ( { keyword, synonyms } ) )
```

Reply:
Nice! I like it.